src/main/java/com/csc366/project/entity/Contract.java

```diff
@@ -0,0 +1,162 @@
+package com.csc366.project.entity;
+
+import java.io.Serializable;
+import java.time.LocalDate;
+import java.util.Objects;
+import java.util.StringJoiner;
+
+import javax.persistence.Column;
+import javax.persistence.Embeddable;
+import javax.persistence.EmbeddedId;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
+
+
+@Entity
+@Table(name="contract")
+public class Contract {
+    @Embeddable
+    public static class ContractId implements Serializable {
+        @Column(nullable=false, updatable = false)
+        private String productId;
+
+        @Column(nullable=false, updatable = false)
+        private Long supplierId;
+
+        @Column(nullable=false, updatable = false)
+        private Long locationId;
```

Comment:
I think this implies that a supplier can only ever supply a particular product to a particular location once. It might be better to have an auto-generated id for each contract that is the PK, but that's just my 2 cents

Reply:
You're right. I'll change it to just an auto-generated ID. That simplifies things too.